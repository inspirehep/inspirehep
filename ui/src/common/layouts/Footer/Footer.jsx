import React from 'react';
import { Layout, Row, Col } from 'antd';

import './Footer.scss';
import ExternalLink from '../../components/ExternalLink';


const Footer = () => (
  <Layout.Footer className="__Footer__">
    <Row type="flex" align="bottom" justify="space-between">
      <Col>
        <ExternalLink className="footer-link" href="//invenio-software.org/">Powered by Invenio</ExternalLink>
      </Col>
      <Col>
        <Row type="flex" justify="start">
          <Col>
            <ExternalLink className="footer-link" href="//inspirehep.net/info/general/privacy-policy">
              Privacy Policy
            </ExternalLink>
          </Col>
          <Col>
            <ExternalLink className="footer-link" href="//inspirehep.net/info/general/terms-of-use">
              Terms of Use
            </ExternalLink>
          </Col>
          <Col>
            <ExternalLink className="footer-link" href="mailto:feedback@inspirehep.net">
              Contact
            </ExternalLink>
          </Col>
          <Col>
            <ExternalLink className="footer-link" href="https://twitter.com/inspirehep">
              Twitter
            </ExternalLink>
          </Col>
          <Col>
            <ExternalLink className="footer-link" href="//blog.inspirehep.net">
              Blog
            </ExternalLink>
          </Col>
        </Row>
      </Col>
    </Row>
  </Layout.Footer >
);

export default Footer;
