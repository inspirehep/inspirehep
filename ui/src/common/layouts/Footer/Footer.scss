.__Footer__ {
  z-index: $zindex-footer;
  background-color: $layout-header-background;
}
