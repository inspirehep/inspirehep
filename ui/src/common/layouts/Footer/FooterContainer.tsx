import { connect } from 'react-redux';

import Footer from './Footer';
import { isCataloger } from '../../authorization';

const stateToProps = state => ({
  isCatalogerLoggedIn: isCataloger(state.user.getIn(['data', 'roles'])),
});

export default connect(stateToProps)(Footer);
