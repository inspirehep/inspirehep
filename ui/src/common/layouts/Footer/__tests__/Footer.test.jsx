import { render } from '@testing-library/react';

import Footer from '../Footer';

describe('Footer', () => {
  it('renders when cataloger', () => {
    const { getByRole } = render(<Footer isCatalogerLoggedIn />);
    expect(getByRole('link', { name: 'Holdingpen' })).toBeInTheDocument();
    expect(getByRole('link', { name: 'Author list' })).toBeInTheDocument();
  });

  it('renders when not cataloger', () => {
    const { queryByRole } = render(<Footer isCatalogerLoggedIn={false} />);
    expect(queryByRole('link', { name: 'Holdingpen' })).not.toBeInTheDocument();
    expect(queryByRole('link', { name: 'Backoffice' })).not.toBeInTheDocument();
    expect(
      queryByRole('link', { name: 'Author list' })
    ).not.toBeInTheDocument();
  });
});
