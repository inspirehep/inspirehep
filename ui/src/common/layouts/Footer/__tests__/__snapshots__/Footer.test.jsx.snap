// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<Basic
  className="__Footer__"
  prefixCls="ant-layout-footer"
>
  <Row
    align="bottom"
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col>
      <ExternalLink
        className="footer-link"
        href="//invenio-software.org/"
      >
        Powered by Invenio
      </ExternalLink>
    </Col>
    <Col>
      <Row
        gutter={0}
        justify="start"
        type="flex"
      >
        <Col>
          <ExternalLink
            className="footer-link"
            href="//inspirehep.net/info/general/privacy-policy"
          >
            Privacy Policy
          </ExternalLink>
        </Col>
        <Col>
          <ExternalLink
            className="footer-link"
            href="//inspirehep.net/info/general/terms-of-use"
          >
            Terms of Use
          </ExternalLink>
        </Col>
        <Col>
          <ExternalLink
            className="footer-link"
            href="mailto:feedback@inspirehep.net"
          >
            Contact
          </ExternalLink>
        </Col>
        <Col>
          <ExternalLink
            className="footer-link"
            href="https://twitter.com/inspirehep"
          >
            Twitter
          </ExternalLink>
        </Col>
        <Col>
          <ExternalLink
            className="footer-link"
            href="//blog.inspirehep.net"
          >
            Blog
          </ExternalLink>
        </Col>
      </Row>
    </Col>
  </Row>
</Basic>
`;
